ADXL362 add separate triggers for activity/inactivity

maxd-nordic opened the pull request:
This patch adds separate triggers for activity and inactivity detection
for the ADXL362 accelerometer.
Previously, applications could not distinguish these events.

Signed-off-by: Maximilian Deubel [maximilian.deubel@nordicsemi.no](mailto:maximilian.deubel@nordicsemi.no)

maxd-nordic committed:
sensors: add trigger types for motion detection

This patch adds the new trigger types SENSOR_TRIG_MOTION
and SENSOR_TRIG_STATIONARY to distinguish these common
motion detection events.

Signed-off-by: Maximilian Deubel <maximilian.deubel@nordicsemi.no>

carlescufi commented:
@gmarull @avisconti could you please take a look?

nvlsianpu reviewed (dismissed):
Code looks fine to me.
Need to add records to the release note: one for ADXL362 and second note for sensor API extension.

henrikbrixandersen commented on include/zephyr/drivers/sensor.h:
```diff
@@ -236,6 +236,11 @@ enum sensor_trigger_type {
 	SENSOR_TRIG_FREEFALL,
 
+	/** Trigger fires when motion is detected. */
+	SENSOR_TRIG_INSTANT_MOTION,
```
Why `INSTANT_`?

maxd-nordic replied:
done

MaureenHelm commented on samples/sensor/adxl362/src/main.c:
```diff
@@ -22,8 +22,11 @@ static void trigger_handler(const struct device *dev,
 		break;
-	case SENSOR_TRIG_THRESHOLD:
-		printf("Threshold trigger\n");
+	case SENSOR_TRIG_INSTANT_MOTION:
+		printf("Activity trigger\n");
+		break;
+	case SENSOR_TRIG_STATIONARY:
+		printf("Inactivity trigger\n");
```
Please align the printf strings to match the names of the triggers

maxd-nordic replied:
done

maxd-nordic committed:
ADXL362: Add separate activity/inactivity triggers

This patch modifies the ADXL362 driver to use the new
SENSOR_TRIG_MOTION trigger for activity detection
and SENSOR_TRIG_STATIONARY for inactivity detection.

Signed-off-by: Maximilian Deubel <maximilian.deubel@nordicsemi.no>

maxd-nordic commented on drivers/sensor/adxl362/adxl362.h:
```diff
@@ -196,8 +196,10 @@ struct adxl362_data {
+	sensor_trigger_handler_t inact_th_handler;
+	struct sensor_trigger inact_th_trigger;
+	sensor_trigger_handler_t act_th_handler;
+	struct sensor_trigger act_th_trigger;
```
done
